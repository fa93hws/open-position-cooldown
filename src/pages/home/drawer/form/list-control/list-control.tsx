import * as React from 'react';
import { Switch, IconButton } from '@material-ui/core';
import { AddBox, Delete } from '@material-ui/icons';
import { makeStyles } from '@material-ui/core/styles';

type ListControlProps = {
  removeChecked: boolean;
  onRemoveChange(val: boolean): void;
  onAddClick(): void;
};

const useSwitchStyles = makeStyles({
  sizeSmall: {
    '& $switchBase': {
      padding: 0,
    },
  },
  switchBase: {},
});

const useIconButtonStyles = makeStyles({
  root: {
    padding: 0,
  },
});

export const ListControl = React.memo((props: ListControlProps) => {
  const switchStyles = useSwitchStyles();
  const iconButtonStyles = useIconButtonStyles();
  const UncheckIcon = <Delete color="secondary" />;
  const CheckedIcon = <Delete />;
  const onRemoveChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
    props.onRemoveChange(ev.target.checked);
  };
  return (
    <>
      <Switch
        id="switch"
        size="small"
        color="primary"
        checked={props.removeChecked}
        checkedIcon={CheckedIcon}
        icon={UncheckIcon}
        classes={switchStyles}
        onChange={onRemoveChange}
      />
      <IconButton
        id="add-reason"
        color="primary"
        classes={iconButtonStyles}
        onClick={props.onAddClick}
      >
        <AddBox />
      </IconButton>
    </>
  );
});
