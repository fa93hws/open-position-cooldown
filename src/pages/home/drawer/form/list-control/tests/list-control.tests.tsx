import * as React from 'react';
import { mount } from 'enzyme';

import { ListControl } from '../list-control';

describe('ListControl', () => {
  it('renders the list control when the switch is false', () => {
    expect(
      <div>
        <ListControl
          onAddClick={jest.fn()}
          onRemoveChange={jest.fn()}
          removeChecked={false}
        />
      </div>,
    ).toMatchRenderedSnapshot();
  });

  it('renders the list control when the switch is true', () => {
    expect(
      <div>
        <ListControl
          onAddClick={jest.fn()}
          onRemoveChange={jest.fn()}
          removeChecked={true}
        />
      </div>,
    ).toMatchRenderedSnapshot();
  });

  it('can display remove icon by switch', () => {
    const onRemoveChange = jest.fn();
    const jsxElement = (
      <div>
        <ListControl
          onAddClick={jest.fn()}
          onRemoveChange={onRemoveChange}
          removeChecked={true}
        />
      </div>
    );
    const element = mount(jsxElement);
    element
      .find('input#switch')
      .simulate('change', { target: { checked: true } });
    expect(onRemoveChange).toHaveBeenCalledWith(true);
  });
});
