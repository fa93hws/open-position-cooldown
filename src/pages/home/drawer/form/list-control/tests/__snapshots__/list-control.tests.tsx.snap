// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListControl renders the list control when the switch is false 1`] = `
Array [
  <span
    class="MuiSwitch-root MuiSwitch-sizeSmall makeStyles-sizeSmall-1"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase makeStyles-switchBase-2 MuiSwitch-colorPrimary"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          class="PrivateSwitchBase-input-7 MuiSwitch-input"
          id="switch"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiSwitch-track"
    />
  </span>,
  <button
    class="MuiButtonBase-root MuiIconButton-root makeStyles-root-3 MuiIconButton-colorPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
    </span>
  </button>,
]
`;

exports[`ListControl renders the list control when the switch is true 1`] = `
Array [
  <span
    class="MuiSwitch-root MuiSwitch-sizeSmall makeStyles-sizeSmall-1"
  >
    <span
      aria-disabled="false"
      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-4 MuiSwitch-switchBase makeStyles-switchBase-2 MuiSwitch-colorPrimary PrivateSwitchBase-checked-5 Mui-checked"
    >
      <span
        class="MuiIconButton-label"
      >
        <input
          checked=""
          class="PrivateSwitchBase-input-7 MuiSwitch-input"
          id="switch"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </span>
    <span
      class="MuiSwitch-track"
    />
  </span>,
  <button
    class="MuiButtonBase-root MuiIconButton-root makeStyles-root-3 MuiIconButton-colorPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
        />
      </svg>
    </span>
  </button>,
]
`;
